### ISSUE TITLE
Update Deployment Config Param To Update Entire Instance Config

### ISSUE THREAD

**Issue by nharper285** [microsoft/onefuzz#2208]
Here is how the instance_config vs. the deployment config (what's passed in during deployment) works. 

First, when we wrote the NSG PR, we made it such that the config parameter passed in during deployment only updates the NSG rules, not the entire instance_config. That means that passing an entire instance_config into the deployment param will not create or update other instance_config rules.   

Therefore, in order to properly update the entire instance_config you currently have to update manually via `onefuzz instance_config update ....`

Once it's been updated manually, you can continuously deploy over it with an NSG config that has the correct rules and there will be no change to the other instance_config rules. 

tl;dr, the only way to set most of the instance_config rules is to do it manually. Once it's been done manually, it can be deployed over indefinitely. Only nsg-rules are tied to the config passed in at deployment.  

**Feature Request: I think it'd be perfectly reasonable for us to update the current deployment logic to update the entire instance_config.**
